Sources/SwiftDocC/Infrastructure/External Data/OutOfProcessReferenceResolver.swift

```diff
@@ -762,30 +810,54 @@ extension OutOfProcessReferenceResolver {
                     willEventuallyBeDeprecated: false // This information isn't used anywhere since this node doesn't have its own page, it's just referenced from other pages.
                 )
             })
-        } else {
-            availability = nil
+            availabilityVariants[mainTrait] = availability
+        }
+        
+        var kindVariants = [mainTrait: symbolKind(forNodeKind: kind)]
+        var titleVariants = [mainTrait: title]
+        var subHeadingVariants = [DocumentationDataVariantsTrait: [SymbolGraph.Symbol.DeclarationFragments.Fragment]]()
+        subHeadingVariants[mainTrait] = declarationFragments?.declarationFragments
+        var abstractVariants = [DocumentationDataVariantsTrait: AbstractSection]()
+        abstractVariants[mainTrait] = abstract.map { AbstractSection(paragraph: Paragraph([Text($0)])) }
+        
+        for variant in variants ?? [] {
```

Comment:
Nit: This structure was a little hard for me to follow although I see it makes sense now. Maybe a introductory explanatory comment would be helpful?

Reply:
It might be worth considering refactoring this logic into a convenience initializer for `Symbol`, which takes the parameters provided to this function in some form.